types/fs-extra/index.d.ts

```diff
@@ -1,85 +1,256 @@
-export declare function writeJSON(file: string, object: any, options?: WriteOptions, callback?: (err: Error) => void): void;
-
-export declare function writeJsonSync(file: string, object: any, options?: WriteOptions): void;
-export declare function writeJSONSync(file: string, object: any, options?: WriteOptions): void;
-
-export declare function ensureDir(path: string, callback: (err: Error) => void): void;
-export declare function ensureDirSync(path: string): void;
-
-export declare function ensureFile(path: string, callback: (err: Error) => void): void;
-export declare function ensureFileSync(path: string): void;
-
-export declare function ensureLink(path: string, callback: (err: Error) => void): void;
-export declare function ensureLinkSync(path: string): void;
-
-export declare function ensureSymlink(path: string, callback: (err: Error) => void): void;
-export declare function ensureSymlinkSync(path: string): void;
-
-export declare function emptyDir(path: string, callback?: (err: Error) => void): void;
-export declare function emptyDirSync(path: string): boolean;
+export function readJson(file: string, options?: ReadOptions): Promise<any>;
+export function readJson(file: string, callback: (err: Error, jsonObject: any) => void): void;
+export function readJson(file: string, options: ReadOptions, callback: (err: Error, jsonObject: any) => void): void;
+export function readJSON(file: string, options?: ReadOptions): Promise<any>;
+export function readJSON(file: string, callback: (err: Error, jsonObject: any) => void): void;
+export function readJSON(file: string, options: ReadOptions, callback: (err: Error, jsonObject: any) => void): void;
+
+export function readJsonSync(file: string, options?: ReadOptions): any;
+export function readJSONSync(file: string, options?: ReadOptions): any;
+
+export function remove(dir: string): Promise<void>;
+export function remove(dir: string, callback: (err: Error) => void): void;
+export function removeSync(dir: string): void;
+
+export function writeJSON(file: string, object: any, options?: WriteOptions): Promise<void>;
+export function writeJSON(file: string, object: any, callback: (err: Error) => void): void;
+export function writeJSON(file: string, object: any, options: WriteOptions, callback: (err: Error) => void): void;
+export function writeJson(file: string, object: any, options?: WriteOptions): Promise<void>;
+export function writeJson(file: string, object: any, callback: (err: Error) => void): void;
+export function writeJson(file: string, object: any, options: WriteOptions, callback: (err: Error) => void): void;
+
+export function writeJsonSync(file: string, object: any, options?: WriteOptions): void;
+export function writeJSONSync(file: string, object: any, options?: WriteOptions): void;
+
+export function ensureDir(path: string): Promise<void>;
+export function ensureDir(path: string, callback: (err: Error) => void): void;
+export function ensureDirSync(path: string): void;
```

Comment:
`ensureDir*` methods are aliases for `mkdirp`/`mkdirs`. They technically can take MkdirOptions.

However, I'm not sure if MkdirOptions should be included at all since the fact that any of these methods can take options isn't documented. https://github.com/jprichardson/node-fs-extra/blob/master/docs/ensureDir.md